single_kernel_mongo/managers/tls.py

```diff
@@ -233,153 +173,237 @@ def enable_certificates_for_unit(self):
             statuses_state=self.state.statuses,
             component_name=self.charm.name,
         )
+
+        if self.is_waiting_for_a_cert():
+            logger.info("Still waiting for a certificate, delaying restart.")
+            return
+
         try:
             self.dependent.restart_charm_services(force=True)
         except WorkloadServiceError as e:
             # TODO should we defer or just error
             logger.error("An exception occurred when starting mongod agent, error: %s.", str(e))
             return
 
-    def delete_certificates_from_workload(self):
+    def delete_certificates_from_workload(self, internal: bool) -> None:
         """Deletes the certificates from the workload."""
-        logger.info("Deleting TLS certificates from filesystem")
+        logger.info(
+            f"Deleting {TLSType.PEER.value if internal else TLSType.CLIENT.value} TLS certificates from filesystem"
+        )
 
-        for file in self.workload.paths.tls_files:
+        path = (
+            self.workload.paths.tls_peer_files if internal else self.workload.paths.tls_client_files
+        )
+        for file in path:
             if self.workload.exists(file):
                 self.workload.delete(file)
 
         if self.substrate == Substrates.VM:
             return
 
-        local_keyfile_file = self.state.paths.ext_pem_file
-        local_ca_file = self.state.paths.ext_ca_file
-        for file in (local_keyfile_file, local_ca_file):
-            if file.exists() and file.is_file():
-                file.unlink()
+        if not internal:
+            local_keyfile_file = self.state.paths.ext_pem_file
+            local_ca_file = self.state.paths.ext_ca_file
+            for file in (local_keyfile_file, local_ca_file):
+                if file.exists() and file.is_file():
+                    file.unlink()
 
-    def push_tls_files_to_workload(self) -> None:
+    def push_tls_files_to_workload(self, internal: bool) -> None:
         """Pushes the TLS files on the workload."""
-        external_ca, external_pem = self.get_tls_files(internal=False)
-        internal_ca, internal_pem = self.get_tls_files(internal=True)
-        if external_ca is not None:
-            self.workload.write(self.workload.paths.ext_ca_file, external_ca)
-        if external_pem is not None:
-            self.workload.write(self.workload.paths.ext_pem_file, external_pem)
-        if internal_ca is not None:
-            self.workload.write(self.workload.paths.int_ca_file, internal_ca)
-        if internal_pem is not None:
-            self.workload.write(self.workload.paths.int_pem_file, internal_pem)
+        logger.info(
+            f"Pushing {TLSType.PEER.value if internal else TLSType.CLIENT.value} TLS certificates to filesystem"
+        )
+        ca, pem = self.get_tls_file_contents(internal=internal)
+
+        if internal:
+            if ca is not None:
```

Comment:
nit: for this one and the following ones
```suggestion
            if ca:
```